src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -200,7 +200,17 @@ public bool IsEmpty(DocumentId documentId)
                 foreach (var documentId in result.DocumentIds)
                 {
                     var document = project.GetDocument(documentId);
-                    Contract.ThrowIfNull(document);
+                    if (document == null)
+                    {
+                        // it can happen with build synchronization since, in build case, 
+                        // we don't have actual snapshot (we have no idea what sources out of proc build has picked up)
+                        // so we might be out of sync.
+                        // example of such cases will be changing anything about solution while building is going on.
+                        // it can be user explict actions such as unloading project, deleting a file, but also it can be 
+                        // something project system or roslyn workspace does such as populating workspace right after
+                        // solution is loaded.
+                        continue;
```

Comment:
one example of the race is file being changed while build is going on, if that file has errors from build, its error location can be out of sync. and we will still show those out of sync errors to users.

it will be soon corrected to right one by live error though. if it is opened file, automatically, if it is closed file, once user interact with VS.